scripts/scripts.js

```diff
@@ -615,6 +615,52 @@ export async function loadScript(url, attrs = {}) {
   return loadingPromise;
 }
 
+export const handleModalClick = async (a, modalFragmentBlock) => {
+  a.addEventListener('click', async (e) => {
+    e.preventDefault();
+    const target = e.currentTarget.parentElement.querySelector('a');
+    if (!target) return;
+    const { path } = target.dataset;
+    const modalId = target.dataset.modal;
+    const elem = document.getElementById(modalId);
+    const hasSearchParam = (target.dataset.hasSearchParam === 'true');
+
+    if (!elem || e.target.dataset.hasSearchParam) {
+      if (hasSearchParam) modalFragmentBlock.innerHTML = '';
+      const wrapper = document.createElement('div');
+      wrapper.className = 'modal-wrapper';
+      wrapper.id = modalId;
+      wrapper.dataset.url = target.dataset.url;
+
+      const modal = document.createElement('div');
```

Comment:
Nit: Could we maybe refactor this to use the new-ish `dialog` tag?
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog
https://caniuse.com/?search=dialog

Reply:
yes, makes sense. Logged https://github.com/hlxsites/sunstar-foundation/issues/68